addons/azureSets/azuresets.lua

```diff
@@ -159,39 +199,42 @@ function find_spell_id_by_name(spellname)
     return nil
 end
 
-function set_single_spell(setspell,slot)
-    if windower.ffxi.get_player()['main_job_id'] ~= 16 --[[and windower.ffxi.get_player()['sub_job_id'] ~= 16]] then return nil end
+function set_single_spell(setspell, slot)
+    if not bluJobLevel then return nil end
 
-    local tmpTable = T(get_current_spellset())
-    for key,val in pairs(tmpTable) do
-        if tmpTable[key]:lower() == setspell then
+    update_current_spellset()
+    for key, val in pairs(currentSpellSet) do
+        if currentSpellSet[key]:lower() == setspell then
             error('That spell is already set.')
             return
         end
     end
     if tonumber(slot) < 10 then slot = '0'..slot end
     --insert spell add code here
-    for spell in spells:it() do
-        if spell['english']:lower() == setspell then
-            --This is where single spell setting code goes.
-            --Need to set by spell id rather than name.
-            windower.ffxi.set_blue_magic_spell(spell['id'], tonumber(slot))
+    local spellId = find_spell_id_by_name(setspell)
+    if spellId then
+        local verified = verify_and_set_spell(spellId, tonumber(slot))
+        if verified then
             windower.send_command('@timers c "Blue Magic Cooldown" 60 up')
-            tmpTable['slot'..slot] = setspell
+            currentSpellSet['slot'..slot] = setspell
         end
     end
-    tmpTable = nil
 end
 
-function get_current_spellset()
-    if windower.ffxi.get_player().main_job_id ~= 16 then return nil end
-    return T(windower.ffxi.get_mjob_data().spells)
+function update_current_spellset(player)
+    if not get_blu_job_data then
+        currentSpellSet = nil
+        return nil
+    end
+
+    currentSpellSet = T(get_blu_job_data().spells)
```

Comment:
I set it up that way to ensure that the most recent spellset was always the one currently loaded. If the command to set the spell fails then it will attempt to do it again.